### ISSUE TITLE
Commits pushed by `pick_to_branch` tend to get lost

### ISSUE THREAD

**Issue by DDvO** [openssl/tools#141]
Since a couple of months it happens to me rather often 
(for instance, today for https://github.com/openssl/openssl/pull/20298 and https://github.com/openssl/openssl/pull/20257) that 
after a successful `pick-to-branch` the commits do not really end up (or at least do not prevail) in the upstream destination.

As a workaround, I have to check after a couple of minutes and in case the commits were silently ignored, push them manually.

**Comment by t8m** [comment]
I think this must be something with your repo setup. Do you have remote branches for those stable branches on github.openssl.org?

**Comment by DDvO** [comment]
If it is an issue with the repo setup, I wonder why it works when pushing manually.
Here is mine:

```
git remote -v
origin	git@github.com:siemens/openssl.git (fetch)
origin	git@github.com:siemens/openssl.git (push)
upstream	git@github.openssl.org:openssl/openssl.git (fetch)
upstream	git@github.openssl.org:openssl/openssl.git (push)
```
```
git branch -vv
* master           8b6bbcaa7b [origin/master] ...
  openssl-3.0      44e816bd54 [origin/openssl-3.0] ...
  openssl-3.1      a3446d12d8 [origin/openssl-3.1] ....
```

**Comment by t8m** [comment]
Hmm that looks correct. However I really can't see what could be wrong with the script and it also never happened to me.

**Comment by DDvO** [comment]
Thanks for your comments.
The strange thing is that at first the commits appear to be present upstream, but after a few minutes they are lost.

**Comment by t8m** [comment]
It would look exactly like that if you pushed the master commit to github.openssl.org and the cherry-picked commits to github.com/openssl/openssl. But that cannot happen unless you have the remote branch on github.com/openssl/openssl.

**Comment by DDvO** [comment]
Oh I meanwhile understand what went wrong for me using `pick_to_branch`:
The push went to the right branch, but in the wrong repo/remote.

I've provided #142 to add a warning if so.

**Comment by DDvO** [comment]
> If it is an issue with the repo setup, I wonder why it works when pushing manually. Here is mine:
> 
> ```
> git remote -v
> origin	git@github.com:siemens/openssl.git (fetch)
> origin	git@github.com:siemens/openssl.git (push)
> upstream	git@github.openssl.org:openssl/openssl.git (fetch)
> upstream	git@github.openssl.org:openssl/openssl.git (push)
> ```
> 
> ```
> git branch -vv
> * master               8b6bbcaa7b [origin/master] ...
>   openssl-3.0          44e816bd54 [origin/openssl-3.0] ...
>   openssl-3.1          a3446d12d8 [origin/openssl-3.1] ....
>   OpenSSL_1_1_1-stable 3f499b24f3 [upstream/OpenSSL_1_1_1-stable] ...
> ```

So the tracking info of my 3.0 and 3.1 branches should have looked like this:
```
  openssl-3.0          ba56257067 [upstream/openssl-3.0] ...
  openssl-3.1          54be1fe86b [upstream/openssl-3.1] ...
```

**Comment by DDvO** [comment]
Closing since the problem was on my (user's) end and meanwhile a warning has been implemented via #142.
